TracyOpenCL.hpp

```diff
@@ -35,12 +35,28 @@ using TracyCLCtx = void*;
 
 #include <atomic>
 #include <cassert>
+#include <sstream>
 
 #include "Tracy.hpp"
 #include "client/TracyCallstack.hpp"
 #include "client/TracyProfiler.hpp"
 #include "common/TracyAlloc.hpp"
 
+#define TRACY_CL_TO_STRING_INDIRECT(T) #T
+#define TRACY_CL_TO_STRING(T) TRACY_CL_TO_STRING_INDIRECT(T)
+#define TRACY_CL_ASSERT(p) if(!(p)) {                                                         \
+    TracyMessageL( "TRACY_CL_ASSERT failed on " __FILE__ ":" TRACY_CL_TO_STRING(__LINE__) );  \
+    assert(false && "TRACY_CL_ASSERT failed");                                                \
+}
+#define TRACY_CL_CHECK_ERROR(err) if(err != CL_SUCCESS) {                                 \
+    std::ostringstream oss;                                                               \
+    oss << ("TRACY_CL_CHECK_ERROR failed on " __FILE__ ":" TRACY_CL_TO_STRING(__LINE__))  \
```

Comment:
I thought that the whole goal of using streams would be to avoid creating strings in a convoluted way like this. Why not simply

```c++
oss << "TRACY_CL_CHECK_ERROR failed on " << __FILE__ << ":" << __LINE__ << ": error code" << err
```
you shouldn't need to use a stringify macro when using streams.

Reply:
Done